### DISCUSSION TITLE
How to properly setup Google Consent Mode V2

### DISCUSSION THREAD

**Discussion by Josepdal** [johannschopplich/nuxt-gtag#78] [Q&A]
Hello, I have some questions related to Consent Mode.

- Do I have to activate anything on the Analytics page?
- Do I have to activate anything on the Google Tag Manager page?
- Since I am using a third-party cookie manager, It is a bit difficult for me to access to nuxt functions. So, would it be enough to activate the consent mode using the following code:

```
initCommands: [
      // Setup up consent mode
      ['consent', 'default', {
        ad_user_data: 'denied',
        ad_personalization: 'denied',
        ad_storage: 'denied',
        analytics_storage: 'denied',
        wait_for_update: 500,
      }]
    ]
```

Or if I do not grant permission ("granted") after user accepts cookies I will lose a lot of data?
I am only using Analytics, not ads or anything related.

Thanks in advance!

**Comment by Josepdal** [comment]
@johannschopplich 👀 could you help? Thanks!

**Reply by johannschopplich** [reply]
Sorry, I don't have the resources to help you at the moment. Doesn't the README instructions don't work for you? Once you get it working, feel free to post your results here and maybe even open a PR with some improvements in the README. Thanks in advance. 🙏

**Comment by Priyanshu10230** [comment]
To properly set up Google Consent Mode:
- **Understand Consent Mode:** Consent Mode adjusts Google tags' behavior based on user consent for cookies and data collection, ensuring compliance with privacy regulations like GDPR and CCPA.
 - **[Integrate a Consent Management Platform (CMP)](https://www.tatvic.com/blog/consent-mode-v2-implementation/?utm_campaign=forum_answer&utm_source=github&utm_medium=blog&utm_term=integrate%20a%20consent%20management%20platform%20(cmp)):** Use a CMP to collect and manage user consent effectively. Google Consent Mode is not a replacement for a CMP.
 •  **Implement Consent Signals:**
 •  ad_storage: For marketing and advertising cookies.
 •  analytics_storage: For Google Analytics cookies.
 •  functionality_storage: For functional cookies like language preferences.
 •  personalization_storage: For personalized user experiences.
 •  security_storage: For security and authentication.
 •  **Upgrade to Consent Mode V2:** Includes additional signals like ad_user_data and ad_personalization for enhanced consent management and compliance with new regulations.
- **Adopt Conversion Modeling:** V2 uses conversion modeling to estimate conversions for users who do not consent, based on data from consenting users.
- **Ensure Compliance and Maintain Data Integrity:**
 •  Avoid fines and penalties by adhering to privacy regulations.
 •  Preserve data for conversion tracking and audience building by implementing Consent Mode V2.

Proper implementation ensures compliance with privacy regulations and maximizes data utility.
